sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample07_MockEventProcessorHandlersTests.cs

```diff
@@ -0,0 +1,244 @@
+
+            using var cancellationSource = new CancellationTokenSource();
+
+            // Applicaition handler cancels events processing after detection of the
+            // exact error with exact operation.
+
+            Task processErrorHandler(ProcessErrorEventArgs args)
+            {
+                if (args.Exception.Message.Equals("Example exception") && args.Operation.Equals("Example operation"))
+                {
+                    cancellationSource.Cancel();
+                }
+
+                return Task.CompletedTask;
+            }
+
+            // Initialize the args with the specific operation name and exception message.
+
+            var exception = new Exception("Example exception");
+            var eventArgs = new ProcessErrorEventArgs("0", "Example operation", exception);
+
+            // Execute the handler and validate that the specific event arguments were detected
+            // and the cancellation has been triggered.
+
+            await processErrorHandler(eventArgs);
+            Assert.IsTrue(cancellationSource.IsCancellationRequested);
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task InitializeEventHandlerCancellationRequest()
+        {
+            #region Snippet:EventHubs_Processor_Sample07_InitializeEventHandlerCancellationRequest
+
+            // Application handler honors cancellation and doesn't set the
+            // default starting postition.
+
+            Task initializeEventHandler(PartitionInitializingEventArgs args)
+            {
+                if (args.CancellationToken.IsCancellationRequested)
+                {
+                    return Task.CompletedTask;
+                }
+
+                args.DefaultStartingPosition = EventPosition.Latest;
+                return Task.CompletedTask;
+            }
+
+            // Initialize the args with a cancelled token and a starting position
+            // different than what gets set if the handler completes.
+
+            using var cancellationSource = new CancellationTokenSource();
+            cancellationSource.Cancel();
+
+            var defaultStartingPosition = EventPosition.Earliest;
+            var eventArgs = new PartitionInitializingEventArgs("0", defaultStartingPosition, cancellationSource.Token);
+
+            // Execute the handler and validate that cancellation was respected.
+
+            await initializeEventHandler(eventArgs);
+            Assert.That(eventArgs.DefaultStartingPosition, Is.EqualTo(defaultStartingPosition));
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task CloseEventHandlerOwnershipLostReason()
```

Comment:
I'm having the same thought here that I did with the error handler.   Should we consider removing this and just talk about "all event handlers can be tested by creating their args and calling them directly" - with the initialization being the general example and the process event handler being the one with special needs?

Reply:
That would kind of lead us to the markdown talking about:

1) the general case
1) showing the initialization hander example
1) discussing that the event handler is a bit different
1) showing the event handler example

Reply:
After going through all comments, I think it might be better to group some event handling in one sample. 

For example if we want to demonstrate how mocking of the offset could be used, we can capture it in the _catch_ block of the **normal event handler** and use it in **init even handler**. All in one sample. Is it not too  much?

Reply:
Then Checkpointing sample could just demonstrate mocking of the custom UpdateCheckpointAsync function. Quite compact... Will it be enough?

Reply:
I'll remove Error and Close handlers' samples. They are confusing indeed.

Reply:
Thanks for your comments @jsquire, I'll disappear again for a bit, but I'll get back next week. I think, I can make Checkpointing sample a bit more interesting with offset or sequence number.

Reply:
> We could capture offset in the _catch_ block of the **normal event handler** and use it in **init even handler**. All in one sample. Is it not too much?

If I understand what you're suggesting, that may give the wrong impression.   The default starting position is considered only when no checkpoints exist for the partition.   The approach that you're pitching would be a nice feature for recovery in the case where an application is intentionally not checkpointing - but we see a lot of confusion on the default being an override.

Reply:
I have updated the tests according to our conversation. Please have a look and thanks again for your feedback! There still can be some silly things due to the lack of the knowledge, but I hope we're slowly getting there.

Reply:
Hi @jsquire, I think the Checkpoint (first) example is now a bit more mature and a step closer to be considered as good for the sample. Please have a look.